File: aoc_2022/jour02##/2.c
#include <stdbool.h>
#include <stdio.h>

int getValue(char c) {
	if (c == 'X')
		return 1;
	else if (c == 'Y')
		return 2;
	else
		return 3;
}

int getWin(char c1, char c2) {
	if (c1 == 'X')
		if (c2 == 'B')
			return 0;
		else if (c2 == 'A')
			return 3;
		else
			return 6;
	else if (c1 == 'Y')
		if (c2 == 'C')
			return 0;
		else if (c2 == 'B')
			return 3;
		else
			return 6;
	else
		if (c2 == 'A')
			return 0;
		else if (c2 == 'C')
			return 3;
		else
			return 6;
}

char newMe(char c1, char win) {
	if (win == 'Y') {
		if (c1 == 'A')
			return 'X';
		else if (c1 == 'B')
			return 'Y';
		else
			return 'Z';
	}
	else if (win == 'Z') {
		if (c1 == 'A')
			return 'Y';
		else if (c1 == 'B')
			return 'Z';
		else
			return 'X';
	}
	else
		if (c1 == 'A')
			return 'Z';
		else if (c1 == 'B')
			return 'X';
		else
			return 'Y';
}

int main() {
	//read file
	FILE *file;
	file = fopen("input.txt", "r");
	char line[100]; 
	fgets(line, 100, file);
	char me;
	char opponent;
	int score = 0;
	while (!feof(file)) {
		opponent = line[0];
		me = newMe(opponent, line[2]);
		int itemScore = getValue(me);
		int winScore = getWin(me, opponent);
		score += itemScore;
		score += winScore;
		printf("%s", line);
		printf("%d + %d\n", itemScore, winScore);
		fgets(line, 100, file);
	}
	printf("\nTotal score : %d\n", score);
	return 0;
}

File: aoc_2022/jour02##/2.py
lines = [l.split() for l in open("input.txt")]
print(lines)

def win(c):
    match c:
        case 'X':
            return 0
        case 'Y':
            return 3
        case 'Z':
            return 6

def value(c1, c2):
    match c1, c2:
        case 'A', 'X':
            return 3
        case 'A', 'Y':
            return 1
        case 'A', 'Z':
            return 2
        case 'B', 'X':
            return 1
        case 'B', 'Y':
            return 2
        case 'B', 'Z':
            return 3
        case 'C', 'X':
            return 2
        case 'C', 'Y':
            return 3
        case 'C', 'Z':
            return 1


scores = [value(a, b) + win(b) for (a, b) in lines]
print(sum(scores))
